x-pack/plugins/apm/common/search_strategies/latency_correlations/types.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  HistogramItem,
+  RawResponseBase,
+  SearchStrategyClientParams,
+} from '../types';
+
+export interface LatencyCorrelation {
+  correlation: number;
+  fieldName: string;
+  fieldValue: string;
+  histogram: HistogramItem[];
+  ksTest: number;
+}
+
+// Basic type guard for array of LatencyCorrelation
+export function isLatencyCorrelations(
+  arg: unknown
+): arg is LatencyCorrelation[] {
+  return Array.isArray(arg) && arg.length > 0;
```

Comment:
Would be better to have a check that at least the first item has `correlation`, `histogram`, and `ksTest` value

Reply:
FYI Decided the type guard was overkill and changed it back to inline checks in c52158b89a0.